src/couch_replicator.erl

```diff
@@ -305,3 +296,86 @@ doc_from_db(RepDb, DocId, UserCtx) ->
          {not_found, _Reason} ->
             {error, not_found}
     end.
+
+
+-spec check_authorization(rep_id(), #user_ctx{}) ->
+    authorized | unauthorized | not_found.
+check_authorization(RepId, #user_ctx{name = Name, roles = Roles}) ->
+    case couch_replicator_scheduler:rep_state(RepId) of
```

Comment:
True there is a race condition but it works like before, that can be a separate bug for a different pr?  We'd have to push authorization even further down the stack, into the doc_processor gen_server to avoid it?